### PR TITLE
Add gatekeeper version streams

### PR DESCRIPTION
**Pull request by tcnghia**
* "gatekeeper-3.12 & 3.13 version streams"

### CODE REVIEW

File: gatekeeper-3.12.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package:
+  name: gatekeeper-3.12
+  version: 3.12.0
+  epoch: 0
+  description: Gatekeeper - Policy Controller for Kubernetes
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    provides:
+      - gatekeeper=3.12
```
Comment (mattmoor) [inline gatekeeper-3.12.yaml:10]: I think we had a `3.12`, so this needs to be `3.12.999` (see `consul-1.15` for example)

File: gatekeeper-3.12.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  - runs: |
+      cd gatekeeper
+      FRAMEWORKS_VERSION=$(go list -f '{{ .Version }}' -m github.com/open-policy-agent/frameworks/constraint)
+      OPA_VERSION=$(go list -f '{{ .Version }}' -m github.com/open-policy-agent/opa)
+      CGO_ENABLED=0 GO111MODULE=on go build -mod vendor -a -ldflags "-s -w -X github.com/open-policy-agent/gatekeeper/pkg/version.Version=v${{package.version}} -X main.frameworksVersion=${FRAMEWORKS_VERSION} -X main.opaVersion=${OPA_VERSION}" -o manager
+      mkdir -p ${{targets.destdir}}/usr/bin
+      install -Dm755 ./manager ${{targets.destdir}}/usr/bin/manager
+
+subpackages:
+  - name: gatekeeper-compat
```
Comment (mattmoor) [inline gatekeeper-3.12.yaml:39]:
```suggestion
  - name: ${{package.name}}-compat
```

File: gatekeeper-3.12.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      provides:
+        - gatekeeper-compat=3.12
+
+update:
+  enabled: true
+  github:
+    identifier: open-policy-agent/gatekeeper
+    strip-prefix: v
+    use-tag: true
+    tag-filter: v
```
Comment (mattmoor) [inline gatekeeper-3.12.yaml:54]:
```suggestion
    tag-filter: v3.12.
```

File: gatekeeper-3.13.yaml [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+  name: gatekeeper-3.13
   version: 3.13.0
-  epoch: 1
+  epoch: 0
   description: Gatekeeper - Policy Controller for Kubernetes
   copyright:
     - license: Apache-2.0
+  dependencies:
+    provides:
+      - gatekeeper=3.13
```
Comment (mattmoor) [inline gatekeeper-3.13.yaml:10]: same re: `.999`

File: gatekeeper-3.12.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+subpackages:
+  - name: gatekeeper-compat
+    pipeline:
+      - runs: |
+          mkdir -p ${{targets.subpkgdir}}
+          ln -sf /usr/bin/manager ${{targets.subpkgdir}}/manager
+    dependencies:
+      provides:
+        - gatekeeper-compat=3.12
```
Comment (mattmoor) [inline gatekeeper-3.12.yaml:46]: `.999`

File: gatekeeper-3.13.yaml [hunk 6]
```diff
@@ -38,6 +41,9 @@ subpackages:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}
           ln -sf /usr/bin/manager ${{targets.subpkgdir}}/manager
+    dependencies:
+      provides:
+        - gatekeeper-compat=3.13
```
Comment (mattmoor) [inline gatekeeper-3.13.yaml:46]: Same re: versioned `${{package.name}}` for subpackage names and `.999` here

File: gatekeeper-3.13.yaml [hunk 7]
```diff
@@ -38,6 +41,9 @@ subpackages:
...
 
 update:
   enabled: true
```
Comment (mattmoor) [inline gatekeeper-3.13.yaml:49]: same re: `tag-filter`

File: gatekeeper-3.12.yaml (hunk 1 above)
Comment (tcnghia) [reply gatekeeper-3.12.yaml:10]: Done

File: gatekeeper-3.12.yaml (hunk 2 above)
Comment (tcnghia) [reply gatekeeper-3.12.yaml]: Done

File: gatekeeper-3.12.yaml (hunk 3 above)
Comment (tcnghia) [reply gatekeeper-3.12.yaml]: Done

File: gatekeeper-3.13.yaml (hunk 4 above)
Comment (tcnghia) [reply gatekeeper-3.13.yaml]: Done

File: gatekeeper-3.12.yaml (hunk 5 above)
Comment (tcnghia) [reply gatekeeper-3.12.yaml]: Done

File: gatekeeper-3.13.yaml (hunk 6 above)
Comment (tcnghia) [reply gatekeeper-3.13.yaml]: Done

File: gatekeeper-3.13.yaml (hunk 7 above)
Comment (tcnghia) [reply gatekeeper-3.13.yaml:49]: Done

File: gatekeeper-3.13.yaml [hunk 8]
```diff
@@ -33,16 +36,19 @@ pipeline:
...
 subpackages:
-  - name: gatekeeper-compat
+  - name: ${{package.name}}-compat
     pipeline:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}
           ln -sf /usr/bin/manager ${{targets.subpkgdir}}/manager
+    dependencies:
+      provides:
+        - ${{package.name}}-compat=3.13.999
```
Comment (mattmoor) [inline gatekeeper-3.13.yaml:46]: This is actually wrong now because `package.name` is `gatekeeper-3.13`.

Same comment for `3.12` above.

Comment (tcnghia) [reply gatekeeper-3.13.yaml]: Done

**Review by mattmoor — APPROVED** [review]

**Pull request merged**